Lib/asyncio/tasks.py

```diff
@@ -23,6 +24,9 @@
 from . import futures
 from .coroutines import coroutine
 
+# Helper to generate new task names
+_counter = itertools.count(1).__next__
```

Comment:
Indeed this seems like an overkill. There's no reason to make Task creation any slower than it has to be; keep in mind that PyPy uses the pure Python version. Just replace this with a simple int increment.

Reply:
This is what the threading module does, and it uses `itertools.count()` because `_counter += 1` is not thread safe. The same applies to asyncio if there are more than one thread running an event loop (since the counter is global).

Reply:
> It is thread safe in Python and will likely always be with or without the GIL.

What output does [this script](https://gist.github.com/agronholm/c06c2dec5eb77c98ab264151fff6178c) give you then?

Reply:
Hmmm... thank you for pointing this out! ;)

To my point: `+=` is actually atomic, but storing the updated number in globals is another operation and we have a race between them. This actually deserves a comment in the code explaining why we use `counter()` and not `+=`.